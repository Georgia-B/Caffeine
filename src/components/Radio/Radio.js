import React from 'react';
import PropTypes from 'prop-types';
import styles from './Radio.module.css';

const Radio = ({ value, label, isSelected, onSelect, type, image = null }) => {
    const className = isSelected ? styles.selected : styles.inputwrapper;
    const content = image !== null ?
        <img src={image.src} alt={image.alt} height={image.height} />
        :
        <span>{label}</span>;

    return (
        <label className={`${className} ${styles[type]}`}>
            {content}
            <input type="radio" name={type} value={value} onClick={onSelect} />
        </label>
    );
};

Radio.propTypes = {
    value: PropTypes.string,
    name: PropTypes.string,
    label: PropTypes.string,
    isSelected: PropTypes.bool,
    onSelect: PropTypes.func,
    image: PropTypes.object,
    type: PropTypes.string,
};

export default Radio;
