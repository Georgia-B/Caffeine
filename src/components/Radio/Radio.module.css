.inputwrapper {
    cursor: pointer;
    background-color: #f1e5cc;
    -webkit-appearance: none;
    outline: none;
    display: flex;
    align-items: center;
    justify-content: center;
}

.inputwrapper > input {
    opacity: 0;
    width: 0;
    margin: 0;
    padding: 0;
}

.inputwrapper:hover {
    background-image: linear-gradient(#f1e5cc, #ecddbd, #e9d7b1);
}

.inputwrapper:focus-within {
    box-shadow: 0 0 4px #91732a;
}

.selected {
    composes: inputwrapper;
    background-color: #ffdb8a;
}

.selected:hover,
.selected:focus {
    background-image: linear-gradient(#ffdb8a, #f5d281, #f1d080);
}

.style {
    width: 100px;
    height: 50px;
    margin: 5px;
    border-radius: 5px; 
}

.milk {
    width: 60px;
    height: 60px;
    border-radius: 30px;
    margin: 10px;
    font-size: 14px;
}

.size {
    width: 80px;
    height: 90px;
    margin: 5px 10px;
    border-radius: 5px;
}