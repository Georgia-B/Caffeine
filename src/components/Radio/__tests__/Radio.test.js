import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderIntl } from '../../../test';
import Radio from '../Radio';

describe('Components | <Radio />', () => {
    afterEach(cleanup);
    it('should render correctly with text', () => {
        const component = renderIntl(<Radio
            value="Hello"
            onSelect={() => true}
            isSelected={false}
            image={null}
            type="test"
        />);

        expect(component.asFragment()).toMatchSnapshot();
    });

    it('should render correctly with image', () => {
        const component = renderIntl(<Radio
            value="Hello"
            onSelect={() => true}
            isSelected={false}
            image={{
                src: '/image.png',
                alt: 'helloimage',
                height: '30',
            }}
            type="test"
        />);

        expect(component.asFragment()).toMatchSnapshot();
    });

    it('should render correctly when selected', () => {
        const component = renderIntl(<Radio
            value="Hello"
            onSelect={() => true}
            isSelected={true}
            image={null}
            type="test"
        />);

        expect(component.asFragment()).toMatchSnapshot();
    });
});
