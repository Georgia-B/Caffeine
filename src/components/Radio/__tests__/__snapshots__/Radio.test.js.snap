// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Radio /> should render correctly when selected 1`] = `
<DocumentFragment>
  <label
    class="selected test"
  >
    <span>
      Hello
    </span>
    <input
      name="test"
      type="radio"
      value="Hello"
    />
  </label>
</DocumentFragment>
`;

exports[`Components | <Radio /> should render correctly with image 1`] = `
<DocumentFragment>
  <label
    class="inputwrapper test"
  >
    <img
      alt="helloimage"
      height="30"
      src="/image.png"
    />
    <input
      name="test"
      type="radio"
      value="Hello"
    />
  </label>
</DocumentFragment>
`;

exports[`Components | <Radio /> should render correctly with text 1`] = `
<DocumentFragment>
  <label
    class="inputwrapper test"
  >
    <span>
      Hello
    </span>
    <input
      name="test"
      type="radio"
      value="Hello"
    />
  </label>
</DocumentFragment>
`;
